### PR TITLE
Task 17

### PR DESCRIPTION
**Pull request by RuslanGabd**

### CODE REVIEW

File: springweb/src/main/java/com/ruslan/controller/UserController.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    private final MapperUser mapperUser;
+
+    @PostMapping("/login")
+    public JwtDto login(@RequestBody UserDto userDto) {
+        return new JwtDto(userService.login(mapperUser.userDtoToUser(userDto)));
+    }
+
+    @PostMapping("/registration")
+    public String addUser(@ModelAttribute("userForm") @Valid User userForm, BindingResult bindingResult, Model model) throws UserExistException {
```
Comment (mrseneca) [inline springweb/src/main/java/com/ruslan/controller/UserController.java]: dto for return type

**Review by mrseneca — APPROVED** [review]
Well done, just check the comment

File: springweb/src/main/java/com/ruslan/userService/UserService.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+    @Override
+    public boolean saveUser(User user) throws UserExistException {
+        if (userRepository.findByUsername(user.getUsername()).isPresent()) {
+            throw new UserExistException(user.getUsername());
+        }
+        Role role = roleRepository.findByName("USER").orElse(null);
+        user.setListRoles(Collections.singletonList(role));
+        userRepository.createUser(user);
+        return true;
```
Comment (mrseneca) [inline springweb/src/main/java/com/ruslan/userService/UserService.java]: User must be saved with crypted password

**Review by mrseneca — CHANGES_REQUESTED** [review]
Found one big issue, need to fix

File: springweb/src/main/java/com/ruslan/userService/UserService.java (hunk 2 above)
Comment (RuslanGabd) [reply springweb/src/main/java/com/ruslan/userService/UserService.java]: @mrseneca  Thank you, I fixed. Could you please review

File: springweb/src/main/java/com/ruslan/controller/UserController.java (hunk 1 above)
Comment (RuslanGabd) [reply springweb/src/main/java/com/ruslan/controller/UserController.java]: fixed

**Review by mrseneca — APPROVED** [review]
Well done

**Pull request merged**